junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/execution/ExtensionValuesStore.java

```diff
@@ -43,6 +44,24 @@ public ExtensionValuesStore(ExtensionValuesStore parentStore) {
 		this.parentStore = parentStore;
 	}
 
+	// Only effects values stored in this instance -- it does not close values in parent stores.
+	public void closeAllStoredAutoCloseableValues() {
+		if (storedValues.isEmpty()) {
+			return;
+		}
```

Comment:
do we need this `if`?

Reply:
The just-in-compiler should generate something similar, I guess. Remove it?

Reply:
I'd say so.